home-manager.nix

```diff
@@ -14,25 +14,84 @@ in
     home.persistence = mkOption {
       default = { };
       type = with types; attrsOf (
-        submodule {
+        submodule ({ name, ... }: {
           options =
             {
               directories = mkOption {
                 type = with types; listOf str;
                 default = [ ];
+                example = [
+                  "Downloads"
+                  "Music"
+                  "Pictures"
+                  "Documents"
+                  "Videos"
+                  "VirtualBox VMs"
+                  ".gnupg"
+                  ".ssh"
+                  ".local/share/keyrings"
+                  ".local/share/direnv"
+                ];
+                description = ''
+                  A list of directories in your home directory that
+                  you want to link to persistent storage.
+                '';
               };
 
               files = mkOption {
                 type = with types; listOf str;
                 default = [ ];
+                example = [
+                  ".screenrc"
+                ];
+                description = ''
+                  A list of files in your home directory you want to
+                  link to persistent storage.
+                '';
+              };
+
+              allowOther = mkOption {
+                type = with types; nullOr bool;
+                default = null;
+                example = true;
+                apply = x:
+                  if x == null then
+                    warn ''
```

Comment:
Is it necessary to display this warning? Why not just default to `false`?

Reply:
It's not strictly necessary, no. It's mainly that I think most people would want to set it to `true`, but probably don't know it. Essentially, we can't set it to the sane default, since it clashes with the default NixOS configuration.